pkg/deploy/kustomizer.go

```diff
@@ -0,0 +1,209 @@
+func manageResource(
+	ctx context.Context,
+	cli client.Client,
+	scheme *runtime.Scheme,
+	res *resource.Resource,
+	ownerInstance *llamav1alpha1.LlamaStackDistribution,
+) error {
+	// prevent the controller from trying to apply changes to its own CR
+	if res.GetKind() == llamav1alpha1.LlamaStackDistributionKind && res.GetName() == ownerInstance.Name && res.GetNamespace() == ownerInstance.Namespace {
+		return nil
+	}
+
+	u := &unstructured.Unstructured{}
+	if err := yaml.Unmarshal([]byte(res.MustYaml()), u); err != nil {
+		return fmt.Errorf("failed to unmarshal resource: %w", err)
+	}
+
+	kGvk := res.GetGvk()
+	gvk := schema.GroupVersionKind{
+		Group:   kGvk.Group,
+		Version: kGvk.Version,
+		Kind:    kGvk.Kind,
+	}
+
+	found := u.DeepCopy()
+	err := cli.Get(ctx, client.ObjectKeyFromObject(u), found)
+	if err != nil {
+		if !k8serr.IsNotFound(err) {
+			return fmt.Errorf("failed to get resource: %w", err)
+		}
+		return createResource(ctx, cli, u, ownerInstance, scheme, gvk)
+	}
+	return patchResource(ctx, cli, u, found, ownerInstance)
+}
+
+// createResource creates a new resource, setting an owner reference only if it's namespace-scoped.
+func createResource(
+	ctx context.Context,
+	cli client.Client,
+	obj *unstructured.Unstructured,
+	ownerInstance *llamav1alpha1.LlamaStackDistribution,
+	scheme *runtime.Scheme,
+	gvk schema.GroupVersionKind,
+) error {
+	// Check if the resource is cluster-scoped (like a ClusterRole) to avoid
+	// incorrectly setting a namespace-bound owner reference on it.
+	isClusterScoped, err := isClusterScoped(cli.RESTMapper(), gvk)
+	if err != nil {
+		return fmt.Errorf("failed to determine resource scope: %w", err)
+	}
+	if !isClusterScoped {
+		if err := ctrl.SetControllerReference(ownerInstance, obj, scheme); err != nil {
+			return fmt.Errorf("failed to set controller reference for %s: %w", gvk.Kind, err)
+		}
+	}
+	return cli.Create(ctx, obj)
+}
+
+// isClusterScoped checks if a given GVK refers to a cluster-scoped resource.
+func isClusterScoped(mapper meta.RESTMapper, gvk schema.GroupVersionKind) (bool, error) {
+	mapping, err := mapper.RESTMapping(gvk.GroupKind(), gvk.Version)
+	if err != nil {
+		return false, fmt.Errorf("failed to get REST mapping for GVK %v: %w", gvk, err)
+	}
+	return mapping.Scope.Name() == meta.RESTScopeNameRoot, nil
+}
+
+// patchResource patches an existing resource, but only if we own it.
+func patchResource(ctx context.Context, cli client.Client, desired, existing *unstructured.Unstructured, ownerInstance *llamav1alpha1.LlamaStackDistribution) error {
+	// Critical safety check to prevent the operator from "stealing" or
+	// overwriting a resource that was created by another user or controller.
+	isOwner := false
+	for _, ref := range existing.GetOwnerReferences() {
+		if ref.UID == ownerInstance.GetUID() {
+			isOwner = true
+			break
+		}
+	}
+	if !isOwner {
+		return fmt.Errorf("failed to patch resource %s/%s because it is not owned by this instance",
```

Comment:
If we don't own the resource, we should not return error. Just need to return nil. We can just log this error.